js/client/index.js

```diff
@@ -0,0 +1,164 @@
+
+// Generic Query class implementation
+class Query {
+    // constructor takes in the query number(index) to set two of the queries and reference
+    constructor(index) {
+        this.query1 = fetcher.query1(index);
+        this.query2 = fetcher.query2(index);
+        this.reference = fetcher.ref(index);
+    }
+    
+    // this function usually obtains the parameters needed for the runQuery function except for few cases
+    async getParams() {
+        // skip when query is null
+        if(this.query1 == null) {
+            return;
+        }
+
+        // run the query
+        const data = await queryData(this.query1);
+        // check response mismatch
+        if (data.hasOwnProperty('dataquery') == false) {
+            console.log(`dataquery key not found in the response of the query:\n${this.query1}\n`);
+            failures += 1;
+            return false;
+        }
+        // check empty response
+        if (data.dataquery.length < 0) {
+            console.log(`Empty response returned from Dgraph for query:\n${this.query1}\n`);
+            failures += 1;
+            return false;
+        }
+        // extract the data from object arrays to arrays
+        this.params = data.dataquery.map(element => element[this.reference]);
+        successes += 1;
+        return true;
+    }
+
+    // runQuery function 
+    async runQuery() {
+        // skip when query is null
+        if(this.query2 == null) {
+            return;
+        }
+
+        let data;
+        // run the query - if params is defined, pass variables
+        if (this.params == undefined) {
+            data = await queryData(this.query1);
+        } else {
+            data = await queryData(this.query1, { "$var": this.params });
+        }
+        // check response mismatch
+        if (data.hasOwnProperty('dataquery') == false) {
+            console.log(`dataquery key not found in the response of the query:\n${this.query2}\n`);
+            failures += 1;
+            return false;
+        }
+        // check empty response
+        if (data.dataquery.length < 0) {
+            console.log(`Empty response returned from Dgraph for query:\n${this.query2}\n`);
+            failures += 1;
+            return false;
+        }
+        successes += 1;
+        return true;
+    }
+}
+
+// Creating query instances
+const queryOne = new Query(1);
+const queryTwo = new Query(2);
+const queryThree = new Query(3);
+const queryFour = new Query(4);
+const queryFive = new Query(5);
+const querySix = new Query(6);
+const querySeven = new Query(7);
+const queryEight = new Query(8);
+const queryNine = new Query(9);
+
+// Query tweet data in Dgraph
+async function queryData(query, vars) {
+    // create a transaction
+    const txn = dgraphClient.newTxn({ readOnly: true });
+    // create a response object
+    let response;
+    // if variable is defined 
+    if (vars == undefined) {
+        response = await txn.query(query);
```

Comment:
`return (await txn.query(...)).getJson();` will have same result without a temporary variable named response

And I won't be surprised if queryWithVars handles undefined `vars` correctly.